tests/beaker_tests/cisco_interface/test_l3_ethernet.rb

```diff
@@ -0,0 +1,356 @@
+# necessary.
+def generate_tests_hash(agent) # rubocop:disable Metrics/MethodLength
+  # 'tests' hash
+  # Top-level keys set by caller:
+  # tests[:master] - the master object
+  # tests[:agent] - the agent object
+  #
+  tests = {
+    master: master,
+    agent:  agent,
+  }
+
+  if platform == 'nexus'
+    interface_name = 'ethernet1/4'
+  elsif platform == 'ios_xr'
+    interface_name = 'gigabitethernet0/0/0/1'
+  end
+
+  # tests[id] keys set by caller and used by test_harness_common:
+  #
+  # tests[id] keys set by caller:
+  # tests[id][:desc] - a string to use with logs & debugs
+  # tests[id][:manifest] - the complete manifest, as used by test_harness_common
+  # tests[id][:resource] - a hash of expected states, used by test_resource
+  # tests[id][:resource_cmd] - 'puppet resource' cmd to use with test_resource
+  # tests[id][:ensure] - (Optional) set to :present or :absent before calling
+  # tests[id][:code] - (Optional) override the default exit code in some tests.
+  #
+  # These keys are local use only and not used by test_harness_common:
+  #
+  # tests[id][:manifest_props] - This is essentially a master list of properties
+  #   that permits re-use of the properties for both :present and :absent tests
+  #   without destroying the list
+  # tests[id][:resource_props] - This is essentially a master hash of properties
+  #   that permits re-use of the properties for both :present and :absent tests
+  #   without destroying the hash
+  # tests[id][:title_pattern] - (Optional) defines the manifest title.
+  #   Can be used w/ :af for mixed title/af testing. If mixing, :af values will
+  #   be merged with title values and override any duplicates. If omitted,
+  #   :title_pattern will be set to 'id'.
+  # tests[id][:af] - (Optional) defines the address-family values.
+  #   Must use :title_pattern if :af is not specified. Useful for testing mixed
+  #   title/af manifests
+  #
+  tests['preclean'] = {
+    title_pattern:  interface_name,
+    manifest_props: "
+      ipv4_address        => 'default',
+    ",
+    code:           [0, 2],
+    resource_props: {},
+  }
+
+  tests['default_properties'] = {
+    title_pattern:      interface_name,
+    default_values:     {
+      'description'    => nil,
+      'ipv4_proxy_arp' => 'false',
+      'ipv4_redirects' => platform == 'nexus' ? 'true' : 'false',
+      'mtu'            => platform == 'nexus' ? '1500' : '1514',
+      'vrf'            => nil,
+    },
+    non_default_values: {
+      'ipv4_address'        => "'192.168.1.1'",
+      'ipv4_netmask_length' => "'16'",
+      'shutdown'            => 'false',
+    },
+  }
+
+  if platform == 'nexus'
+    tests['default_properties'][:non_default_values].merge!(
+      'switchport_mode' => "'disabled'",
+    )
+    tests['default_properties'][:default_values].merge!(
+      'speed'                        => "'auto'",
+      'duplex'                       => "'auto'",
```

Comment:
This feels a bit off to me. No doubt this is because of the fact that we're interacting with the Puppet CLI, so it's easier to treat everything as a string, and strings as double-strings, but it would be nice to be able to leave off the quoting and define things as their actual types (and have the helper function handle the stringification as needed):
```ruby
    tests['default_properties'][:default_values].merge!(
      'speed'                        => 'auto',
      'duplex'                       => 'auto',
      'switchport_autostate_exclude' => false,
      'switchport_vtp'               => false,
    )
```

Something for a future US, perhaps.